samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -315,4 +315,48 @@ public void onBarrierError(String version, Throwable t) {
       stop();
     }
   }
+
+  /// listener to handle session expiration
+  public class ZkSessionStateChangedListener implements IZkStateListener {
+
+    @Override
+    public void handleStateChanged(Watcher.Event.KeeperState state)
+        throws Exception {
+      if (state == Watcher.Event.KeeperState.Expired) {
+        // if the session has expired it means that all the registration's ephemeral nodes are gone.
+        LOG.warn("Got session expired event for processor=" + processorId);
```

Comment:
In general yes, but the way it is implemented in zkClient- this call will be made ONLY when a new session is setup, so it is sufficient to do it in the new session handler.